test/oauth-server/service/getOpenIDConfiguration.go

```diff
@@ -16,5 +16,8 @@ func (requestHandler *RequestHandler) getOpenIDConfiguration(w http.ResponseWrit
 		JWKSURL:     requestHandler.getDomain() + uri.JWKs,
 	}
 
-	jsonResponseWriter(w, v)
+	err := jsonResponseWriter(w, v)
+	if err != nil {
+		writeError(w, err, http.StatusBadRequest)
```

Comment:
when w.Write fails you can only log error. it is not recommended to call twice w.WriteHeader